quic/transport/ngtcp2/stream/closedstate.nim

```diff
@@ -48,17 +49,18 @@ method leave*(state: ClosedStream) =
 method read*(state: ClosedStream): Future[seq[byte]] {.async.} =
   if not state.frameSorter.isComplete():
     let incomingFut = state.remaining.get()
-    let timeoutFut = state.connection.timeout.expired()
-    let raceFut = await race(incomingFut, state.cancelRead, timeoutFut)
+    let raceFut = await race(state.cancelRead, state.timedOut, incomingFut)
     if raceFut == incomingFut:
       result = await incomingFut
       allowMoreIncomingBytes(state.stream, state.connection, result.len.uint64)
     else:
       incomingFut.cancelSoon()
       raise
-        if raceFut == timeoutFut:
-          newException(StreamError, "stream timed out")
+        if raceFut == state.timedOut:
+          state.cancelRead.cancelSoon()
```

Comment:
not sure about `state.cancelRead.cancelSoon()` here.

also it strange that we need to do any `reading` in this state. because it feels like now we need additional state to transition to `really closed state` in this else branch.